.laminas-ci/pre-install.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+WORKING_DIRECTORY=$2
+JOB=$3
+PHP_VERSION=$(echo "${JOB}" | jq -r '.php')
+
+cd "$TMPDIR" || exit 1;
+
+apt install -y make g++
+
+pecl channel-update pecl.php.net
+
+# Install lzf
+pecl install lzf || exit 1
+echo "extension=lzf.so" >> /etc/php/"${PHP_VERSION}"/cli/php.ini
+
+# Install rar
+pecl install rar
+echo "extension=rar.so" >> /etc/php/"${PHP_VERSION}"/cli/php.ini
+
+# Install snappy
+git clone --recursive --depth=1 https://github.com/kjdev/php-ext-snappy.git
+cd php-ext-snappy || exit 1;
+phpize
+./configure
+make
+make install
+
+echo "extension=snappy.so" >> /etc/php/"${PHP_VERSION}"/cli/php.ini
+
+# Debug output
+php --ri lzf
+echo ""
+php --ri rar
+echo ""
+php --ri snappy
+echo ""
```

Comment:
TBH, this seems like a rabbit hole of dependency upgrades: are we sure we want to go down the road of custom extensions? :|

Reply:
I'm totally game for deprecating all 3 for removal in 3.0 - The reason for the patch was to figure out if it was even possible for these extensions to be installed on recent versions of PHP - turns out it _is_ possible. I can't imagine use is widespread though.

Shall I deprecate these adapters now? The `rar` one is particularly bad…

Reply:
I'll stick it on the agenda perhaps too.

Reply:
#79 and https://github.com/laminas/technical-steering-committee/pull/128